media/videostitcher/get_live_ad_tag_detail_test.go

```diff
@@ -0,0 +1,104 @@
+// limitations under the License
+
+package videostitcher
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestGetLiveAdTagDetail(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	var buf bytes.Buffer
+	uuid, err := getUUID()
+	if err != nil {
+		t.Fatalf("getUUID err: %v", err)
+	}
+	slateID := fmt.Sprintf("%s-%s", slateIDPrefix, uuid)
+	slateName := fmt.Sprintf("projects/%s/locations/%s/slates/%s", tc.ProjectID, location, slateID)
+	liveConfigID := fmt.Sprintf("%s-%s", liveConfigIDPrefix, uuid)
+	liveConfigName := fmt.Sprintf("projects/%s/locations/%s/liveConfigs/%s", tc.ProjectID, location, liveConfigID)
+
+	createTestSlate(slateID, t)
+	createTestLiveConfig(slateID, liveConfigID, t)
+	sessionID, playURI := createTestLiveSession(liveConfigID, t)
+
+	// No list or delete methods for live sessions
+
+	// Ad tag details
+
+	// To get ad tag details, you need to curl the main manifest and
+	// a rendition first. This supplies media player information to the API.
+	//
+	// Curl the playURI first. The last line of the response will contain a
+	// renditions location. Curl the live session name with the rendition
+	// location appended.
+
+	renditions, err := curlPlayURI(playURI)
+	if err != nil {
+		t.Fatalf("curlPlayURI err: %v", err)
+	}
+
+	// playURI will be in the following format:
+	// https://videostitcher.googleapis.com/v1/projects/{project}/locations/{location}/liveSessions/{session-id}/manifest.m3u8?signature=...
+	// Replace manifest.m3u8?signature=... with the renditions location.
+
+	err = curlRendition(playURI, renditions[0])
+	if err != nil {
+		t.Fatalf("curlRendition err: %v", err)
+	}
+
+	// List the ad tag details for a given live session. This is the only way to get an
+	// ad tag detail.
+	adTagDetailsNamePrefix := fmt.Sprintf("/locations/%s/liveSessions/%s/liveAdTagDetails/", location, sessionID)
+	if err := listLiveAdTagDetails(&buf, tc.ProjectID, sessionID); err != nil {
+		t.Fatalf("listLiveAdTagDetails got err: %v", err)
+	}
+	got := buf.String()
+
+	if !strings.Contains(got, adTagDetailsNamePrefix) {
+		t.Fatalf("listLiveAdTagDetails got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, adTagDetailsNamePrefix)
+	}
+	strSlice := strings.Split(got, "/")
+	adTagDetailsID := strSlice[len(strSlice)-1]
+	adTagDetailsID = strings.TrimRight(adTagDetailsID, "\n")
+	adTagDetailsName := fmt.Sprintf("/locations/%s/liveSessions/%s/liveAdTagDetails/%s", location, sessionID, adTagDetailsID)
+	if !strings.Contains(got, adTagDetailsName) {
+		t.Errorf("listLiveAdTagDetails got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, adTagDetailsName)
+	}
+	buf.Reset()
+
+	// Get the specified ad tag detail for a given live session.
+	testutil.Retry(t, 3, 2*time.Second, func(r *testutil.R) {
+		if err := getLiveAdTagDetail(&buf, tc.ProjectID, sessionID, adTagDetailsID); err != nil {
```

Comment:
Did this sample not have tests already?

Reply:
All code samples previously had tests located in `stitcher_test.go` (see https://github.com/GoogleCloudPlatform/golang-samples/pull/2585).

Reply:
Ah okay. Are there tests that need to be removed then? If so please do so.

Reply:
All tests were removed from `stitcher_test.go`. It now only contains test helper functions and package vars. Each test is now located in its own test file.